Update to upstream 1.13.0

andreittr opened the pull request:
### Description of changes

Update Google Test to upstream 1.13.0, as well as clean up build scripts & documentation.

andreittr committed:
Remove obsolete documentation files

Signed-off-by: Andrei Tatar <andrei@unikraft.io>

andreittr committed:
Update to upstream 1.13.0

Signed-off-by: Andrei Tatar <andrei@unikraft.io>

mariasfiraiala reviewed (changes requested):
The changes look good, thanks @andreittr. Regarding the documentation part, however, the README specifies the order of the libraries  which we don't use anymore (and has `newlib` mentioned as well - kinda obsolete).

AFAIK, the latest correct order is

```Makefile
LIBS := $(UK_LIBS)/lib-libcxxabi:$(UK_LIBS)/lib-libcxx:$(UK_LIBS)/lib-compiler-rt:$(UK_LIBS)/lib-libunwind:$(UK_LIBS)/lib-musl:$(UK_LIBS)/lib-googletest
```

Do you mind changing that too @andreittr? Actually the whole README should be updated to have the md format. I can add a commit to achieve that, if you find it too cumbersome.

andreittr commented:
> The changes look good, thanks @andreittr. Regarding the documentation part, however, the README specifies the order of the libraries which we don't use anymore (and has `newlib` mentioned as well - kinda obsolete).
> 
> AFAIK, the latest correct order is
> 
> ```makefile
> LIBS := $(UK_LIBS)/lib-libcxxabi:$(UK_LIBS)/lib-libcxx:$(UK_LIBS)/lib-compiler-rt:$(UK_LIBS)/lib-libunwind:$(UK_LIBS)/lib-musl:$(UK_LIBS)/lib-googletest
> ```
> 
> Do you mind changing that too @andreittr? Actually the whole README should be updated to have the md format. I can add a commit to achieve that, if you find it too cumbersome.

Good point @mariasfiraiala , thanks! I'll add a commit updating the README to be roughly in line with ones in more up-to-date libs. I'll also simplify the lib order paragraph to just "put at the end of the lib list".

andreittr committed:
Update README & CONTRIBUTING

Reformat to Markdowm.
CONTRIBUTING: remove obsolete references, point users at unikraft website
README: clean up & simplify wording, point users at website

Signed-off-by: Andrei Tatar <andrei@unikraft.io>

mariasfiraiala commented on README.md:
```diff
@@ -1,19 +1,12 @@
-Please refer to the `README.md` as well as the documentation in the
-`doc/` subdirectory of the main unikraft repository for further
-information.
+Being a C++ library, libcxx is required.
```
```suggestion
Being a C++ library, `libcxx` is required.
```

mariasfiraiala commented on README.md:
```diff
@@ -1,19 +1,12 @@
-`doc/` subdirectory of the main unikraft repository for further
-information.
+Being a C++ library, libcxx is required.
+This library should come at the end of the dependency list.
```
```suggestion
`lib-googletest` should come at the end of the dependency list.
```

It isn't clear enough what library should come last.

mariasfiraiala reviewed (changes requested):
Thanks @andreittr, 2 last comments from me below. After this, I think we should merge this.

mariasfiraiala reviewed (approved):
All good now, thanks a lot for the effort, @andreittr!

Reviewed-by: Maria Sfiraiala <maria.sfiraiala@gmail.com>

razvand reviewed (approved):
Approved-by: Razvan Deaconescu <razvand@unikraft.io>
